pkg/infra/baseagent.go

```diff
@@ -222,5 +249,19 @@ func (a *Base) HandleResponse(msg messages.BoolResponseMessage)
 func (a *Base) HandleStateFoodTaken(msg messages.StateFoodTakenMessage)                  {}
 func (a *Base) HandleStateHP(msg messages.StateHPMessage)                                {}
 func (a *Base) HandleStateIntendedFoodTaken(msg messages.StateIntendedFoodIntakeMessage) {}
-func (a *Base) HandleProposeTreaty(msg messages.ProposeTreatyMessage)                    {}
-func (a *Base) HandleTreatyResponse(msg messages.TreatyResponseMessage)                  {}
+
+func (a *Base) HandleProposeTreaty(msg messages.ProposeTreatyMessage) {
+	treatyProposal := msg.Treaty()
+	// always add treaty to active list if proposing
+	a.AddToActiveTreaties(treatyProposal)
+}
```

Comment:
```suggestion
func (a *Base) HandleProposeTreaty(msg messages.ProposeTreatyMessage) {}
```

Reply:
Called when an agent receives a ProposeTreatyMessage

Reply:
this is probably wrong then, not sure what correct behaviour should be

Reply:
Inside this function you would decide whether you want to accept the proposed treaty and then send the appropriate response.